File: android/src/main/java/com/azify/utils/DynamicRoute.java
package com.azify.utils;

import com.azify.processors.Config;

public class DynamicRoute {

  private String getPathUrl(String target, String defaultPathUrl) {
    String processorPathUrl = Config.ProcessorPathURL.get(target);

    if (processorPathUrl == null) {
      return defaultPathUrl;
    } else if (processorPathUrl.trim().isEmpty()) {
      return defaultPathUrl;
    }

    if (processorPathUrl.contains("/:")) {
      processorPathUrl = processorPathUrl.replaceAll("/:([a-zA-Z0-9_]+)", "/" + Config.ProcessId);
    }

    return processorPathUrl;
  }

  public String getPathUrlEnrollment3d(String target) {
    return this.getPathUrl(target, "/Process/" + Config.ProcessId + "/Enrollment3d");
  }

  public String getPathUrlIdScanOnly(String target) {
    return this.getPathUrl(target, "/Process/" + Config.ProcessId + "/idscan-only");
  }

  public String getPathUrlMatch3d2dIdScan(String target) {
    return this.getPathUrl(target, "/Process/" + Config.ProcessId + "/Match3d2dIdScan");
  }

  public String getPathUrlMatch3d3d(String target) {
    return this.getPathUrl(target, "/Process/" + Config.ProcessId + "/Match3d3d");
  }
}

File: ios/Utilities/DynamicRoute.swift
//
//  DynamicRoute.swift
//  Pods
//
//  Created by Daniel Sansão Araldi on 11/08/25.
//

import Foundation

class DynamicRoute: NSObject {

  private func getPathUrl(target: String, defaultPathUrl: String) -> String {
    guard let processorPathUrl = (Config.ProcessorPathURL[target]) else {
      return defaultPathUrl
    }

    if processorPathUrl.trimmingCharacters(in: .whitespaces).isEmpty {
      return defaultPathUrl
    }

    var resultUrl = processorPathUrl

    if processorPathUrl.contains("/:") {
      let pattern = "/:([a-zA-Z0-9_]+)"
      let regex = try! NSRegularExpression(pattern: pattern, options: [])
      let range = NSRange(location: 0, length: processorPathUrl.count)

      resultUrl = regex.stringByReplacingMatches(
        in: processorPathUrl,
        options: [],
        range: range,
        withTemplate: "/\(Config.ProcessId ?? "")"
      )
    }

    return resultUrl
  }

  func getPathUrlEnrollment3d(target: String) -> String {
    return getPathUrl(
      target: target,
      defaultPathUrl: "/Process/\(Config.ProcessId ?? "")/Enrollment3d")
  }

  func getPathUrlMatch3d2dIdScan(target: String) -> String {
    return getPathUrl(
      target: target,
      defaultPathUrl: "/Process/\(Config.ProcessId ?? "")/Match3d2dIdScan")
  }

  func getPathUrlMatch3d3d(target: String) -> String {
    return getPathUrl(
      target: target, defaultPathUrl: "/Process/\(Config.ProcessId ?? "")/Match3d3d")
  }
}
